vignettes/adrs.Rmd

```diff
@@ -422,34 +455,35 @@ dataset_vignette(
 
 ## Derive Clinical Benefit Parameter {#cb}
 
-Similarly, we now define the source location for this newly derived
-first response date.
+The function `derive_extreme_event()` can then be used to derive the clinical
+benefit parameter, which we define as a patient having had a response or a
+sustained period of time before first `PD`. This could also be known as disease
+control. In this example the "sustained period" has been defined as 42 days
+after randomization date via the `cb_y`[^2] event.
 
+[^2]:
 ```{r}
-resp <- date_source(
-  dataset_name = "adrs",
-  date = ADT,
-  filter = PARAMCD == "RSP" & AVALC == "Y"
-)
+cb_y
 ```
-
-The function `derive_param_clinbenefit()` can then be used to derive the
-clinical benefit parameter, which we define as a patient having had a
-response or a sustained period of time before first `PD`. This could
-also be known as disease control. In this example the "sustained period"
-has been defined as 42 days after randomization date, using the
-`ref_start_window` argument.
-
 ```{r}
 adrs <- adrs %>%
-  derive_param_clinbenefit(
-    dataset_adsl = adsl,
-    filter_source = PARAMCD == "OVR" & ANL01FL == "Y",
-    source_resp = resp,
-    source_pd = pd,
-    source_datasets = list(adrs = adrs),
-    reference_date = RANDDT,
-    ref_start_window = 42,
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(desc(AVALC), ADT, PARAMCD),
+    mode = "first",
+    events = list(
+      event(
```

Comment:
why is this not using `rsp_y` directly?

Reply:
for me it fits closer with your new style of building these derivations up with the events passed as source objects so i would say worth to update for a cleaner and more consistent story

Reply:
OK, I have updated it.